ee/clickhouse/queries/test/test_paths.py

```diff
@@ -258,3 +259,271 @@ def test_path_by_funnel(self):
+            [
+                {"source": "1_/screen1", "target": "2_/screen2", "value": 1, "average_conversion_time": ONE_MINUTE},
+                {"source": "2_/screen2", "target": "3_/screen3", "value": 1, "average_conversion_time": 2 * ONE_MINUTE},
+            ],
+        )
+
+        filter = filter.with_data({"include_event_types": ["custom_event"]})
+        response = ClickhousePathsNew(team=self.team, filter=filter).run(team=self.team, filter=filter)
+
+        self.assertEqual(
+            response,
+            [
+                {"source": "1_/custom1", "target": "2_/custom2", "value": 1, "average_conversion_time": ONE_MINUTE},
+                {"source": "2_/custom2", "target": "3_/custom3", "value": 1, "average_conversion_time": 2 * ONE_MINUTE},
+            ],
+        )
+
+        filter = filter.with_data({"include_event_types": [], "include_custom_events": ["/custom1", "/custom2"]})
+        response = ClickhousePathsNew(team=self.team, filter=filter).run(team=self.team, filter=filter)
+
+        self.assertEqual(
+            response,
+            [{"source": "1_/custom1", "target": "2_/custom2", "value": 1, "average_conversion_time": ONE_MINUTE},],
+        )
+
+        filter = filter.with_data({"include_event_types": [], "include_custom_events": ["/custom3", "blah"]})
+        response = ClickhousePathsNew(team=self.team, filter=filter).run(team=self.team, filter=filter)
+
+        self.assertEqual(
+            response, [],
+        )
+
+        filter = filter.with_data(
+            {"include_event_types": ["$pageview", "$screen", "custom_event"], "include_custom_events": []}
+        )
+        response = ClickhousePathsNew(team=self.team, filter=filter).run(team=self.team, filter=filter)
+
+        self.assertEqual(
+            response,
+            [
+                {"source": "1_/1", "target": "2_/2", "value": 1, "average_conversion_time": ONE_MINUTE},
+                {"source": "1_/custom1", "target": "2_/custom2", "value": 1, "average_conversion_time": ONE_MINUTE},
+                {"source": "1_/screen1", "target": "2_/screen2", "value": 1, "average_conversion_time": ONE_MINUTE},
+                {"source": "2_/2", "target": "3_/3", "value": 1, "average_conversion_time": 2 * ONE_MINUTE},
+                {"source": "2_/custom2", "target": "3_/custom3", "value": 1, "average_conversion_time": 2 * ONE_MINUTE},
+                {"source": "2_/screen2", "target": "3_/screen3", "value": 1, "average_conversion_time": 2 * ONE_MINUTE},
+            ],
+        )
+
+        filter = filter.with_data(
+            {
+                "include_event_types": ["$pageview", "$screen", "custom_event"],
+                "include_custom_events": [],
+                "exclude_events": ["/custom1", "$pageview"],
+            }
+        )
+        # TODO: сделать проверку на пустые списки # <- CoPilot changed languages suddenly LOL
+        # real TODO: don't allow including $pageview and excluding $pageview at the same time
```

Comment:
TODO before merge

Reply:
Also, LOL, smart-ish:

![image](https://user-images.githubusercontent.com/7115141/130793607-53e84688-3cc3-4bf5-a9bf-cb7aba29892b.png)

Reply:
Also, excluding specific URLs instead? If we want to do that, then it's better to change API to use `Entity` like elsewhere instead of just the event name.